easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -177,6 +177,24 @@ def det_pylibdir(plat_specific=False, python_cmd=None):
     return pylibdir
 
 
+def det_pip_version():
```

Comment:
I'd refactor this to only do the required work: 
```suggestion
def det_pip_version():
 """Determine version of currently active 'pip' command."""

    out, _ = run_cmd("pip --version", verbose=False, simple=False, trace=False)

    pip_version_regex = re.compile('^pip ([0-9.]+)')
    res = pip_version_regex.search(out)
    if res:
        return res.group(1)
    else:
        log = fancylogger.getLogger('det_pylibdir', fname=False)
        log.warning("Failed to determine pip version from '%s' using pattern '%s'", out, pip_version_regex.pattern)
        return None
```

Reply:
I'll enhance the function a bit to use more logging instead...